components/use-paid-mutation.js

```diff
@@ -99,14 +134,15 @@ export function usePaidMutation (mutation,
         // the action is pessimistic
         try {
           // wait for the invoice to be paid
-          await waitForPayment(invoice, { alwaysShowQROnFailure: true, persistOnNavigate, waitFor })
+          // returns the invoice that was paid since it might have been updated via retries
+          invoice = await waitForPayment(invoice, { alwaysShowQROnFailure: true, persistOnNavigate, waitFor })
           if (!response.result) {
             // if the mutation didn't return any data, ie pessimistic, we need to fetch it
             const { data: { paidAction } } = await getPaidAction({ variables: { invoiceId: parseInt(invoice.id) } })
             // create new data object
             // ( hmac is only returned on invoice creation so we need to add it back to the data )
             data = {
-              [Object.keys(data)[0]]: {
+              [dataKey]: {
                 ...paidAction,
                 invoice: { ...paidAction.invoice, hmac: invoice.hmac }
```

Comment:
this probably does no longer correctly set the hmac since `invoice` is no longer the original invoice